### PR TITLE
chore(proto): Add go package path to proto files (V2 migration)

### PR DESCRIPTION
**Pull request by mtodor**
## Description

This PR is adding the path to proto files for the go package. This was done before 
with #10609, but it didn't work as expected. That PR was revered with #10676.

Providing a proper relative path should output files in the correct locations.

This PR changes:
- all proto files
- all generated `.pb.go` files (but only `fileDescriptor_`, not actual code)

**Note**

I made the following changes:
```
find ./proto/api/v1 -type f -name '*.proto' | xargs -I {} sed -i "" 's/option.*go_package.*/option go_package = ".\/api\/v1;v1";/' {}
find ./proto/api/v2 -type f -name '*.proto' | xargs -I {} sed -i "" 's/option.*go_package.*/option go_package = ".\/api\/v2;v2";/' {}
find ./proto/api/integrations -type f -name '*.proto' | xargs -I {} sed -i "" 's/option.*go_package.*/option go_package = ".\/api\/integrations;integrations";/' {}
find ./proto/storage -type f -name '*.proto' | xargs -I {} sed -i "" 's/option.*go_package.*/option go_package = ".\/storage;storage";/' {}
find ./proto/test -type f -name '*.proto' | xargs -I {} sed -i "" 's/option.*go_package.*/option go_package = ".\/test;test";/' {}
find ./proto/tools/local-sensor -type f -name '*.proto' | xargs -I {} sed -i "" 's/option.*go_package.*/option go_package = ".\/tools\/local-sensor;localSensor";/' {}
find ./proto/internalapi/central -type f -name '*.proto' | xargs -I {} sed -i "" 's/option.*go_package.*/option go_package = ".\/internalapi\/central;central";/' {}
find ./proto/internalapi/compliance -type f -name '*.proto' | xargs -I {} sed -i "" 's/option.*go_package.*/option go_package = ".\/internalapi\/compliance;compliance";/' {}
find ./proto/internalapi/sensor -type f -name '*.proto' | xargs -I {} sed -i "" 's/option.*go_package.*/option go_package = ".\/internalapi\/sensor;sensor";/' {}
find ./proto/internalapi/wrapper -type f -name '*.proto' | xargs -I {} sed -i "" 's/option.*go_package.*/option go_package = ".\/internalapi\/wrapper;wrapper";/' {}
find ./proto/internalapi/scanner/v4 -type f -name '*.proto' | xargs -I {} sed -i "" 's/option.*go_package.*/option go_package = ".\/internalapi\/scanner\/v4;v4";/' {}
```

And regenerated files with `make proto-generated-srcs`.

## Checklist
- [x] Investigated and inspected CI test results
- ~[ ] Unit test and regression tests added~
- ~[ ] Evaluated and added CHANGELOG entry if required~
- ~[ ] Determined and documented upgrade steps~
- ~[ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~

## Testing Performed

- Just CI run
- Checked generated code with a script: https://gist.github.com/mtodor/80eb0f3b5d53d2d369028dd30f198dbc (diff without `fileDescriptor_*`) - it looked good
- Tested also new file in location `./proto/api/v3/user.proto` -> generated files looked good and they have been in correct location for the current generator and protobuf V2

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

File: proto/internalapi/sensor/admission_control.proto [hunk 1]
```diff
@@ -9,7 +9,7 @@ import "storage/deployment.proto";
 import "storage/namespace_metadata.proto";
 import "storage/policy.proto";
 
-option go_package = "sensor";
+option go_package = "./internalapi/central;sensor";
```
Comment (dashrews78) [inline proto/internalapi/sensor/admission_control.proto:12]:
```suggestion
option go_package = "./internalapi/sensor;sensor";
```

File: proto/internalapi/sensor/admission_control_iservice.proto [hunk 2]
```diff
@@ -5,7 +5,7 @@ package sensor;
 import "google/protobuf/empty.proto";
 import "internalapi/sensor/admission_control.proto";
 
-option go_package = "sensor";
+option go_package = "./internalapi/central;sensor";
```
Comment (dashrews78) [inline proto/internalapi/sensor/admission_control_iservice.proto:8]: `sensor`

File: proto/internalapi/sensor/cert_distribution_iservice.proto [hunk 3]
```diff
@@ -4,7 +4,7 @@ package sensor;
 
 import "storage/service_identity.proto";
 
-option go_package = "sensor";
+option go_package = "./internalapi/central;sensor";
```
Comment (dashrews78) [inline proto/internalapi/sensor/cert_distribution_iservice.proto:7]: `sensor`

**Review by dashrews78 — CHANGES_REQUESTED** [review]
All the ones in `./internalapi/sensor` are incorrectly pathed to `./internalapi/central`

**Comment by janisz** [comment]
is it working with v2?

**Comment by dashrews78** [comment]
> is it working with v2?

Seemed to be.  I pulled it down and made a couple of changes in `proto/api/v2/compliance_results_service.proto`.  Files were at least generated in the correct place.  But then I saw the errors below which make sense because all those protos are in `internalapi/sensor`:

```
Removing internalapi/central/deployment_iservice.pb.go because internalapi/central/deployment_iservice.proto does not exist
Removing internalapi/central/network_connection_iservice.pb.go because internalapi/central/network_connection_iservice.proto does not exist
Removing internalapi/central/image_iservice.pb.go because internalapi/central/image_iservice.proto does not exist
Removing internalapi/central/admission_control.pb.go because internalapi/central/admission_control.proto does not exist
Removing internalapi/central/compliance_iservice.pb.go because internalapi/central/compliance_iservice.proto does not exist
Removing internalapi/central/signal_iservice.pb.go because internalapi/central/signal_iservice.proto does not exist
Removing internalapi/central/admission_control_iservice.pb.go because internalapi/central/admission_control_iservice.proto does not exist
Removing internalapi/central/cert_distribution_iservice.pb.go because internalapi/central/cert_distribution_iservice.proto does not exist
Removing internalapi/central/network_connection_info.pb.go because internalapi/central/network_connection_info.proto does not exist
Removing internalapi/central/network_enums.pb.go because internalapi/central/network_enums.proto does not exist
Removing internalapi/central/splunk_alert.pb.go because internalapi/central/splunk_alert.proto does not exist
Removing internalapi/central/collector.pb.go because internalapi/central/collector.proto does not exist
```

**Comment by mtodor** [comment]
@dashrews78 Nice catch! Tnx! I had two wrong ones:
```
find ./proto/internalapi/sensor -type f -name '*.proto' | xargs -I {} sed -i "" 's/option.*go_package.*/option go_package = ".\/internalapi\/central;sensor";/' {}
find ./proto/internalapi/wrapper -type f -name '*.proto' | xargs -I {} sed -i "" 's/option.*go_package.*/option go_package = ".\/internalapi\/central;wrapper";/' {}
```
Classical copy/paste mistake! 😞 

@janisz Yes, it works with V2. I have added info in the migration plan doc.

**Review by dashrews78 — APPROVED** [review]

**Review by janisz — APPROVED** [review]

**Review by janisz — APPROVED** [review]

**Review by rhybrillou — APPROVED** [review]

**Pull request merged**